R/scoring.R

```diff
@@ -85,29 +85,35 @@ get_dec_concepts <- function(res_data) {
 }
 
 get_value_domain <- function(res_data) {
-  
-
-    if (res_data$result$dataElement$name == "NOMATCH" || length(res_data$result$valueDomain) == 0) {
-      tibble::tibble(observedValue = NA, name = NA, id = NA) %>% 
-        dplyr::filter(complete.cases(.))
-    } else {
-      res_data$result$valueDomain %>% 
-        purrr::map(~ list(observedValue = .$observedValue, 
-                          value = .$permissibleValue$value, 
-                          conceptCode = .$permissibleValue$conceptCode)) %>% 
-        purrr::modify_depth(2, ~ ifelse(is.null(.), "", .)) %>%
-        purrr::map_df(tibble::as_tibble) %>%
-        dplyr::mutate(conceptCode = ifelse(conceptCode == "", NA, conceptCode)) %>%
-        dplyr::distinct()
-    }
-
+  if (length(res_data$result$valueDomain) == 0) {
+    tibble::tibble(observedValue = NA, name = NA, id = NA) %>%
+      dplyr::filter(complete.cases(.))
+  } else {
+    res_data$result$valueDomain %>%
+      purrr::map(~ list(
+        observedValue = .$observedValue,
+        value = .$permissibleValue$value,
+        conceptCode = .$permissibleValue$conceptCode
+      )) %>%
+      purrr::modify_depth(2, ~ ifelse(is.null(.), "", .)) %>%
+      purrr::map_df(tibble::as_tibble) %>%
+      dplyr::mutate(conceptCode = ifelse(conceptCode == "", NA, conceptCode)) %>%
+      dplyr::distinct()
+  }
 }
 
 
 find_mismatch_rows <- function(df_a, df_b, col_name = "conceptCode") {
   # TODO: fix logic of this function
-  df_a[[col_name]][!(df_a[[col_name]] %in% df_b[[col_name]])]
+  #df_a[[col_name]][!(df_a[[col_name]] %in% df_b[[col_name]])]
```

Comment:
I wasn't sure about this change, but I trust your judgement. 🙂 If you're confident that the basic behavior is the same, feel free to merge.

Reply:
Ah, right! I made this change for the case where a mismatched col_name is a valid value for another row, e.g.

**minimal goldstandard vd**
```
> test_gold <- tibble(observedValue=c("YES", "NO"), value=c("NOMATCH", "nah"), conceptCode=c(NA,"ncit:000"))
> test_gold
# A tibble: 2 x 3
  observedValue value   conceptCode
  <chr>         <chr>   <chr>      
1 YES           NOMATCH NA         
2 NO            nah     ncit:000 
```

**minimal submission vd**
```
> test_sub <- tibble(observedValue=c("YES", "NO"), value=c("NOMATCH", "nah"), conceptCode=c(NA,NA))
> test_sub
# A tibble: 2 x 3
  observedValue value   conceptCode
  <chr>         <chr>   <lgl>      
1 YES           NOMATCH NA         
2 NO            nah     NA  
```

My understanding is that there should be one mismatch row expected (`""`) based on `conceptCode`, but since `NA` is a valid value for the first row, the `NA` of the second row got a pass:

```
> find_mismatch_rows(test_sub, test_gold)
logical(0)
``` 

I've tested with a bunch of other scenarios and I am 99% certain the basic functionality is the same otherwise 😆